config/routes.rb

```diff
@@ -152,6 +152,7 @@
       post 'structure', to: 'master_files#set_structure', constraints: { format: 'json' }
       delete 'structure', to: 'master_files#delete_structure', constraints: { format: 'json' }
       post 'move'
+      get 'transcript/:t_id/:filename', to: 'master_files#transcript'
```

Comment:
It doesn't look like you're using the `:filename` route component in the controller action.  Could it be dropped?
I think that would make the url be more standard which might make the testing a little more straight forward as well.

Reply:
Yes, `:filename` is not used in the controller action, but this url is being used in the player side to identify the file type with respect to the file extension and it is used to decide which parsing action to take. That's why I included this in there. Maybe I can use `Content-type` in the response in the player side instead of filename (which is more stable), so that the url looks much cleaner?